include/swift/Option/Options.td

```diff
@@ -321,6 +321,16 @@ def warn_swift3_objc_inference : Flag<["-"], "warn-swift3-objc-inference">,
   Alias<warn_swift3_objc_inference_complete>,
   Flags<[FrontendOption, DoesNotAffectIncrementalBuild, HelpHidden]>;
 
+def Rpass_EQ : Joined<["-"], "Rpass=">,
+  Flags<[FrontendOption]>,
+  HelpText<"Report performed transformations by optimization passes whose "
+           "name matches the given POSIX regular expression">;
+
+def Rpass_missed_EQ : Joined<["-"], "Rpass-missed=">,
+  Flags<[FrontendOption]>,
+  HelpText<"Report missed transformations by optimization passes whose "
+           "name matches the given POSIX regular expression">;
```

Comment:
Why make this a single pattern rather than allow multiple arguments?

Reply:
The main motivation was to keep it identical to clang.  That said, I don't think I ever found a good use for the regex capability other than specifying -Rpass=. to enable all passes.  This however could be better achieved by making the value optional and enabling all passes via -Rpass.

I guess one future use case could be to prefix pass names with the name of the IR they operate on.  Then you could enable all SIL remarks for example with -Rpass=sil-?

Reply:
Do we expect -Rpass to control both LLVM and SIL passes? @gottesmm?

Reply:
I was thinking yes, again by generalizing from the situation in LLVM.  In LLVM, it already controls passes on two IRs (LLVM IR and the Machine IR).

Exposing LLVM/MIR level remarks one way or another should be interesting for Swift too.  For example, we could see that a loop was sufficiently simplified by SIL passes in order to allow it to be vectorized by LLVM.  I am fine if we want a different mechanism to enable that mode though.